extensions/standard-processors/processors/SegmentContent.cpp

```diff
@@ -0,0 +1,114 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SegmentContent.h"
+
+#include "core/FlowFile.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "core/Resource.h"
+#include "range/v3/view/split.hpp"
+#include "utils/ProcessorConfigUtils.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+constexpr size_t BUFFER_TARGET_SIZE = 1024;
+
+void SegmentContent::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SegmentContent::onSchedule(core::ProcessContext&, core::ProcessSessionFactory&) {}
+
+namespace {
+std::shared_ptr<core::FlowFile> createSegment(core::ProcessSession& session) {
+  auto first_split = session.create();
+  if (!first_split) { throw Exception(PROCESSOR_EXCEPTION, "Couldn't create FlowFile"); }
+  return first_split;
+}
+
+void updateSplitAttributesAndTransfer(core::ProcessSession& session, const std::vector<std::shared_ptr<core::FlowFile>>& splits, const core::FlowFile& original) {
+  const std::string fragment_identifier_ = utils::IdGenerator::getIdGenerator()->generate().to_string();
+  for (size_t split_i = 0; split_i < splits.size(); ++split_i) {
+    const auto& split = splits[split_i];
+    split->setAttribute(SegmentContent::FragmentCountOutputAttribute.name, std::to_string(splits.size()));
+    split->setAttribute(SegmentContent::FragmentIndexOutputAttribute.name, std::to_string(split_i + 1));  // One based indexing
+    split->setAttribute(SegmentContent::FragmentIdentifierOutputAttribute.name, fragment_identifier_);
+    split->setAttribute(SegmentContent::SegmentOriginalFilenameOutputAttribute.name, original.getAttribute(core::SpecialFlowAttribute::FILENAME).value_or(""));
+    session.transfer(split, SegmentContent::Segments);
+  }
+}
+}  // namespace
+
+void SegmentContent::onTrigger(core::ProcessContext& context, core::ProcessSession& session) {
+  const auto original = session.get();
+  if (!original) {
+    context.yield();
+    return;
+  }
+
+  size_t max_segment_size{};
+  const auto segment_size_str = context.getProperty(SegmentSize, original.get());
+  if (!segment_size_str || !core::DataSizeValue::StringToInt(*segment_size_str, max_segment_size)) {
+    throw Exception(PROCESSOR_EXCEPTION, fmt::format("Invalid Segment Size {}", segment_size_str));
+  }
+
+  const auto ff_content_stream = session.getFlowFileContentStream(*original);
+  if (!ff_content_stream) {
+    throw Exception(PROCESSOR_EXCEPTION, fmt::format("Couldn't access the ContentStream of {}", original->getUUID().to_string()));
+  }
+
+  std::vector<std::byte> buffer;
+  std::vector<std::shared_ptr<core::FlowFile>> segments{};
+
+  size_t current_segment_size = 0;
+  size_t ret{};
```

Comment:
I think `num_bytes_read` would be a better name than `ret`

Reply:
good idea, https://github.com/apache/nifi-minifi-cpp/pull/1879/commits/1c3a18fd8bf777b7c71a7972463e2c8b6f782f82#diff-bcd058832969d67b6e5d615e8ec01b6db147cc3f11ba42a558a0279f74f287cdR75